pkg/config/tls/tls.go

```diff
@@ -0,0 +1,42 @@
+package tls
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/service"
+	"github.com/jaegertracing/jaeger-operator/pkg/util"
+)
+
+// Update will mount the tls secret on the collector pod.
+func Update(jaeger *v1.Jaeger, commonSpec *v1.JaegerCommonSpec, options *[]string) {
+	if viper.GetString("platform") != v1.FlagPlatformOpenShift {
+		return
+	}
+
+	volume := corev1.Volume{
+		Name: configurationVolumeName(jaeger),
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: fmt.Sprintf("%s-tls", service.GetNameForHeadlessCollectorService(jaeger)),
+			},
+		},
+	}
+	volumeMount := corev1.VolumeMount{
+		Name:      configurationVolumeName(jaeger),
+		MountPath: "/etc/config",
+		ReadOnly:  true,
+	}
+	commonSpec.Volumes = append(commonSpec.Volumes, volume)
+	commonSpec.VolumeMounts = append(commonSpec.VolumeMounts, volumeMount)
+	*options = append(*options, "--collector.grpc.tls=true")
```

Comment:
Sorry for not catching this earlier, but this flag has just been deprecated in 1.17:

```console
$ podman run jaegertracing/jaeger-collector:1.17 --help | grep collector.grpc.tls.enabled
      --collector.grpc.tls                              (deprecated) see --collector.grpc.tls.enabled
      --collector.grpc.tls.enabled                      Enable TLS on the server
```